app/src/Page/Shop/product-search.js

```diff
@@ -29,8 +40,27 @@ export async function wHandGet(aReq, aResp) {
 
   const oIsMembEbtEligable = aResp.locals.CredUser?.CdRegEBT === "Approv";
 
-  const { Ct: oCt, Products: oProducts } = await wProducts(aReq.query, oIsMembEbtEligable); //aReq.query includes terms if a term was searched
+  const { Ct: oCt, Products: oProducts } = await wProducts(aReq.query, oIsMembEbtEligable);
   const oDataPage = DataPage(aReq.query, oCt, CtProductPage);
+  // if we know who the user is, pull their favorites and tag each product
+  if (aReq.user) {
+    const memberId = aResp.locals.CredImperUser.IDMemb;
+    if (!memberId) {
+      aResp.status(401);
+      aResp.render("Misc/401");
```

Comment:
we should either not show a link to the favorites pages if they aren't logged in, or we should redirect to login/sign up